sphinx/util/docutils.py

```diff
@@ -464,6 +467,44 @@ def __init__(self, document: nodes.document, builder: "Builder") -> None:
         self.config = builder.config
         self.settings = document.settings
 
+    def dispatch_visit(self, node):
+        """
+        Dispatch node to appropriate visitor method.
+        The priority of visitor method is:
+
+        1. ``self.visit_{parent_node_class}_{node_class}()``
+        2. ``self.visit_{node_class}()``
+        3. ``self.unknown_visit()``
+        """
+        parent_node_name = node.parent.__class__.__name__
+        node_name = node.__class__.__name__
+        method = getattr(self, 'visit_%s_%s' % (parent_node_name, node_name), None)
```

Comment:
It would be better if `visit_%s__%s` (separate with double underscore) to avoid collision with node name like `foo_bar`.